src/Mixins/Expectation.php

```diff
@@ -182,20 +183,87 @@ public function toBeLessThanOrEqual(int|float|DateTimeInterface $expected, strin
     public function toContain(mixed ...$needles): self
     {
         foreach ($needles as $needle) {
-            if (is_string($this->value)) {
-                // @phpstan-ignore-next-line
-                Assert::assertStringContainsString((string) $needle, $this->value);
-            } else {
-                if (! is_iterable($this->value)) {
-                    InvalidExpectationValue::expected('iterable');
+            $this->toContainAtLeast(1, $needle);
+        }
+
+        return $this;
+    }
+
+    /**
+     * Asserts that $needle exists in the value depending on $op $count times.
+     *
+     * @return self<TValue>
+     */
+    public function toContainOp(string $op, int $count, mixed $needle, string $message = ''): self
+    {
+        Assert::assertGreaterThanOrEqual(0, $count);
+
+        $op = str_replace(' ', '', ucwords($op));
```

Comment:
I'm not a big fan of this kind of approach. IMO, this is something that hurts readability. Why not pass the method name without the need of this formatting?